packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts

```diff
@@ -98,14 +98,22 @@ export class ChainAdapter
       const account = await this.getAccount(from)
 
       if (sendMax) {
-        tx.value = bnOrZero(account.balance).minus(gas).toString()
+        try {
+          const val = bnOrZero(account.balance).minus(gas)
+          if (val.isFinite() || val.lte(0)) {
+            throw new Error('CosmosChainAdapter: transaction value is invalid')
```

Comment:
Shouldn't we log the actual wrong tx value?
```suggestion
            throw new Error(`CosmosChainAdapter: transaction value is invalid: ${val.toString()}`)
```